src/robotFactory.js

```diff
@@ -1,10 +1,71 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+    };
+  }
 
-class DeliveryDrone {}
+  goForward(step = 1) {
+    this.coords.y = this.coords.y + step;
+  }
+  goBack(step = 1) {
+    this.coords.y = this.coords.y - step;
+  }
+  goRight(step = 1) {
+    this.coords.x = this.coords.x + step;
+  }
+  goLeft(step = 1) {
+    this.coords.x = this.coords.x - step;
+  }
```

Comment:
You can use assignment right after the math operation

```suggestion
  goForward(step = 1) {
    this.coords.y += step;
  }
  goBack(step = 1) {
    this.coords.y -= step;
  }
  goRight(step = 1) {
    this.coords.x += step;
  }
  goLeft(step = 1) {
    this.coords.x -= step;
  }
```